src/app/features/dashboard-widgets/birthday-dashboard-widget/birthday-dashboard-settings.component/birthday-dashboard-settings.component.ts

```diff
@@ -70,41 +84,111 @@ export class BirthdayDashboardSettingsComponent implements OnInit {
+    this.entityPropertyPairs = Object.entries(entities).map(
+      ([entityType, property]) => ({
+        entityType,
+        property,
+      }),
+    );
+    if (this.entityPropertyPairs.length === 0) {
+      // Always show at least one pair
+      this.entityPropertyPairs.push({
+        entityType: this.availableEntityTypes[0]?.ENTITY_TYPE ?? "",
+        property:
+          this.availablePropertiesMap[
+            this.availableEntityTypes[0]?.ENTITY_TYPE
+          ]?.[0] ?? "",
+      });
+    }
 
-    this.updateAvailableProperties();
+    this.localConfig.threshold = this.formControl.value?.threshold ?? 32;
     this.updateLocalConfig();
+    this.updateAvailableEntityTypesPerRow();
+  }
+
+  availablePropertiesFor(entityType: string): string[] {
+    return this.availablePropertiesMap[entityType] ?? [];
   }
 
-  updateAvailableProperties() {
-    const ctor = this.availableEntityTypes.find(
-      (c) => c.ENTITY_TYPE === this.selectedEntityType,
+  updateAvailableEntityTypesPerRow() {
+    this.availableEntityTypesPerRow = this.entityPropertyPairs.map(
+      (pair, index) => {
+        const usedFields = new Set(
+          this.entityPropertyPairs
+            .filter(
+              (_, i) =>
+                i !== index &&
+                pair.entityType === this.entityPropertyPairs[i].entityType,
+            )
+            .map((p) => p.property),
+        );
+
+        return this.availableEntityTypes.filter((ctor) => {
+          if (ctor.ENTITY_TYPE !== pair.entityType) {
+            return true; // Allow other entity types
+          }
+
+          const availableFields = this.availablePropertiesFor(ctor.ENTITY_TYPE);
+          return availableFields.some(
+            (field) => !usedFields.has(field) || field === pair.property,
+          );
+        });
+      },
     );
-    if (!ctor) {
-      this.availableProperties = [];
-      return;
-    }
-    const schema = getEntitySchema(ctor);
+  }
+
+  get canAddMorePairs(): boolean {
+    return this.availableEntityTypes.some((ctor) => {
+      const entityType = ctor.ENTITY_TYPE;
+      const usedFields = new Set(
+        this.entityPropertyPairs
+          .filter((p) => p.entityType === entityType)
+          .map((p) => p.property),
+      );
+      const availableFields = this.availablePropertiesFor(entityType).filter(
+        (field) => !usedFields.has(field),
+      );
+      return availableFields.length > 0;
+    });
+  }
 
-    this.availableProperties = Array.from(schema.entries())
-      .filter(([_, field]: any) => field.dataType === "date-with-age")
-      .map(([key]) => key);
+  get isFormValid(): boolean {
+    return this.entityPropertyPairs.every(
+      (pair) => pair.entityType && pair.property,
+    );
   }
 
-  onEntityTypeChange(entityType: string) {
-    this.selectedEntityType = entityType;
-    this.updateAvailableProperties();
-    this.selectedProperty = this.availableProperties[0];
+  get hasInvalidPairs(): boolean {
```

Comment:
`isFormValid` and `hasInvalidPairs` look very similar. What's the difference? (Or could we just use a negation to get the same result with just one method?)